----
# **<span style="color:DarkSlateBlue">PROYECTO BOOKING_REVIEWS</span>**
# **<span style="color:DarkSlateBlue">Análisis preliminar</span>**
----

---
---
## <span style="color:gray">**1. Importación de librerías**</span> 📂

In [1]:
# Tratamiento de datos
import numpy as np
import pandas as pd 

# Configuración de ruta
import sys
sys.path.append('../')

# Importación de funciones personalizadas
from src.soporte1_preliminar import eda_preliminar

---
---
## <span style="color:gray">**2. Carga de datos**</span> 📥

In [2]:
booking_raw = pd.read_csv("../data/raw/hotel_booking.csv")

reviews_raw = pd.read_csv("../data/raw/hotel_reviews.csv")

In [3]:
# Configuración para mostrar todas las columnas
pd.set_option('display.max_columns', None)

---
---


----
# <span style="color:DarkSlateBlue">**Desarrollo del proyecto - 1**</span>
----

---
---
## <span style="color:gray">**Exploración inicial de los datasets**</span> 🔎

Antes de comenzar el análisis exploratorio de datos (EDA), es importante realizar una inspección general de los datasets. Este primer contacto con los datos nos permite entender su estructura, el tipo de variables que contienen y detectar posibles problemas como valores faltantes o formatos incorrectos.

Para ello utilizamos varias funciones de Pandas que nos ayudan a obtener una visión rápida y completa del conjunto de datos.

---

## `booking_raw`

In [4]:
# Llamamos a la función eda_preliminar guardada en el archivo de sorporte
eda_preliminar(booking_raw)



👀 Visualización de las primeras y últimas filas



,hotel,is_canceled,lead_time,arrival_date_year,arrival_date_month,arrival_date_week_number,arrival_date_day_of_month,stays_in_weekend_nights,stays_in_week_nights,adults,children,babies,meal,country,market_segment,distribution_channel,is_repeated_guest,previous_cancellations,previous_bookings_not_canceled,reserved_room_type,assigned_room_type,booking_changes,deposit_type,agent,company,days_in_waiting_list,customer_type,adr,required_car_parking_spaces,total_of_special_requests,reservation_status,reservation_status_date,name,email,phone-number,credit_card
0,Resort Hotel,0,342,2015,July,27,1,0,0,2,0.0,0,BB,PRT,Direct,Direct,0,0,0,C,C,3,No Deposit,NaN,NaN,0,Transient,0.0,0,0,Check-Out,2015-07-01,Ernest Barnes,Ernest.Barnes31@outlook.com,669-792-1661,************4322
1,Resort Hotel,0,737,2015,July,27,1,0,0,2,0.0,0,BB,PRT,Direct,Direct,0,0,0,C,C,4,No Deposit,NaN,NaN,0,Transient,0.0,0,0,Check-Out,2015-07-01,Andrea Baker,Andrea_Baker94@aol.com,858-637-6955,************9157
2,Resort Hotel,0,7,2015,July,27,1,0,1,1,0.0,0,BB,GBR,Direct,Direct,0,0,0,A,C,0,No Deposit,NaN,NaN,0,Transient,75.0,0,0,Check-Out,2015-07-02,Rebecca Parker,Rebecca_Parker@comcast.net,652-885-2745,************3734


,hotel,is_canceled,lead_time,arrival_date_year,arrival_date_month,arrival_date_week_number,arrival_date_day_of_month,stays_in_weekend_nights,stays_in_week_nights,adults,children,babies,meal,country,market_segment,distribution_channel,is_repeated_guest,previous_cancellations,previous_bookings_not_canceled,reserved_room_type,assigned_room_type,booking_changes,deposit_type,agent,company,days_in_waiting_list,customer_type,adr,required_car_parking_spaces,total_of_special_requests,reservation_status,reservation_status_date,name,email,phone-number,credit_card
119387,City Hotel,0,34,2017,August,35,31,2,5,2,0.0,0,BB,DEU,Online TA,TA/TO,0,0,0,D,D,0,No Deposit,9.0,NaN,0,Transient,157.71,0,4,Check-Out,2017-09-07,Mary Morales,Mary_Morales@hotmail.com,395-518-4100,************1821
119388,City Hotel,0,109,2017,August,35,31,2,5,2,0.0,0,BB,GBR,Online TA,TA/TO,0,0,0,A,A,0,No Deposit,89.0,NaN,0,Transient,104.40,0,0,Check-Out,2017-09-07,Caroline Conley MD,MD_Caroline@comcast.net,531-528-1017,************7860
119389,City Hotel,0,205,2017,August,35,29,2,7,2,0.0,0,HB,DEU,Online TA,TA/TO,0,0,0,A,A,0,No Deposit,9.0,NaN,0,Transient,151.20,0,2,Check-Out,2017-09-07,Ariana Michael,Ariana_M@xfinity.com,422-804-6403,************4482




------------------------------------------------------------------------


🎲 Visualización de filas aleatorias



,hotel,is_canceled,lead_time,arrival_date_year,arrival_date_month,arrival_date_week_number,arrival_date_day_of_month,stays_in_weekend_nights,stays_in_week_nights,adults,children,babies,meal,country,market_segment,distribution_channel,is_repeated_guest,previous_cancellations,previous_bookings_not_canceled,reserved_room_type,assigned_room_type,booking_changes,deposit_type,agent,company,days_in_waiting_list,customer_type,adr,required_car_parking_spaces,total_of_special_requests,reservation_status,reservation_status_date,name,email,phone-number,credit_card
35617,Resort Hotel,0,276,2017,April,16,22,2,5,1,0.0,0,Undefined,SWE,Groups,TA/TO,0,0,0,A,A,1,No Deposit,527.0,NaN,0,Transient-Party,72.00,0,0,Check-Out,2017-04-29,William Joyce,Joyce_William@outlook.com,338-675-2929,************6481
34998,Resort Hotel,0,207,2017,April,14,6,1,3,1,0.0,0,BB,NZL,Online TA,TA/TO,0,0,0,A,A,0,No Deposit,240.0,NaN,0,Transient,51.00,0,1,Check-Out,2017-04-10,John White,White.John@xfinity.com,930-956-8498,************1375
108765,City Hotel,0,237,2017,March,13,27,1,5,2,0.0,0,BB,RUS,Online TA,TA/TO,0,0,0,A,A,0,No Deposit,9.0,NaN,0,Transient,89.13,0,0,Check-Out,2017-04-02,Kelly Duncan,Duncan.Kelly@comcast.net,547-120-0489,************9907




------------------------------------------------------------------------


📏 Dimensión del DataFrame

Número de filas del DataFrame: 119390
Número de columnas del DataFrame: 36


------------------------------------------------------------------------


🧾 Información general

<class 'pandas.DataFrame'>
RangeIndex: 119390 entries, 0 to 119389
Data columns (total 36 columns):
 #   Column                          Non-Null Count   Dtype  
---  ------                          --------------   -----  
 0   hotel                           119390 non-null  str    
 1   is_canceled                     119390 non-null  int64  
 2   lead_time                       119390 non-null  int64  
 3   arrival_date_year               119390 non-null  int64  
 4   arrival_date_month              119390 non-null  str    
 5   arrival_date_week_number        119390 non-null  int64  
 6   arrival_date_day_of_month       119390 non-null  int64  
 7   stays_in_weekend_nights         119390 non-null  int64  
 

Index(['hotel', 'is_canceled', 'lead_time', 'arrival_date_year',
       'arrival_date_month', 'arrival_date_week_number',
       'arrival_date_day_of_month', 'stays_in_weekend_nights',
       'stays_in_week_nights', 'adults', 'children', 'babies', 'meal',
       'country', 'market_segment', 'distribution_channel',
       'is_repeated_guest', 'previous_cancellations',
       'previous_bookings_not_canceled', 'reserved_room_type',
       'assigned_room_type', 'booking_changes', 'deposit_type', 'agent',
       'company', 'days_in_waiting_list', 'customer_type', 'adr',
       'required_car_parking_spaces', 'total_of_special_requests',
       'reservation_status', 'reservation_status_date', 'name', 'email',
       'phone-number', 'credit_card'],
      dtype='str')



------------------------------------------------------------------------


🔢 Estadísticas descriptivas de las variables numéricas



,count,mean,std,min,25%,50%,75%,max
is_canceled,119390.0,0.370,0.483,0.00,0.00,0.000,1.0,1.0
lead_time,119390.0,104.011,106.863,0.00,18.00,69.000,160.0,737.0
arrival_date_year,119390.0,2016.157,0.707,2015.00,2016.00,2016.000,2017.0,2017.0
arrival_date_week_number,119390.0,27.165,13.605,1.00,16.00,28.000,38.0,53.0
arrival_date_day_of_month,119390.0,15.798,8.781,1.00,8.00,16.000,23.0,31.0
stays_in_weekend_nights,119390.0,0.928,0.999,0.00,0.00,1.000,2.0,19.0
stays_in_week_nights,119390.0,2.500,1.908,0.00,1.00,2.000,3.0,50.0
adults,119390.0,1.856,0.579,0.00,2.00,2.000,2.0,55.0
children,119386.0,0.104,0.399,0.00,0.00,0.000,0.0,10.0
babies,119390.0,0.008,0.097,0.00,0.00,0.000,0.0,10.0




------------------------------------------------------------------------


🔡 Estadísticas descriptivas de las variables categóricas



,count,unique,top,freq
hotel,119390,2,City Hotel,79330
arrival_date_month,119390,12,August,13877
meal,119390,5,BB,92310
country,118902,177,PRT,48590
market_segment,119390,8,Online TA,56477
distribution_channel,119390,5,TA/TO,97870
reserved_room_type,119390,10,A,85994
assigned_room_type,119390,12,A,74053
deposit_type,119390,3,No Deposit,104641
customer_type,119390,4,Transient,89613


### **Informe de EDA preliminar – Dataset `booking_raw`**

**1. Información general del Dataset**

- El Dataset contiene 119390 registros y +30 variables

- Incluye columnas numéricas, categóricas y temporales.

- Se identifican valores nulos en `agent`, `company`, `country` y `children`.

- No presenta registros duplicados → dataset consistente.

**2. Hallazgos por tipo de variable**

Variables numéricas:

- `adr`: alta variabilidad (hasta 5400), indicando posibles outliers.
- `lead_time`: media de 104 días, reflejando reservas con bastante anticipación.
- `is_canceled`: media del 37% → alta tasa de cancelación.

Variables categóricas:

- `adults`, `children`, `babies`: predominan reservas de 1–2 adultos sin niños.
- `previous_cancellations`: valores bajos → pocos clientes reincidentes cancelando.
- Predominan clientes “transient” y reservas online.

**3. Transformaciones clave**

Estandarización:
- `hotel`, `customer_type`, `reservation_status` → minúsculas + snake_case.
- `phone-number` → `phone_number`.

Tipos de datos:
- `is_canceled`, `is_repeated_guest` → bool.
- `reservation_status_date` → datetime.

Eliminación:
- `name`, `email`, `phone_number`, `credit_card` → información del huésped que no es relevante para nuestro análisis.
- `arrival_date_week_number` → ya disponemos de los días, meses y años de llegada. 
- `assigned_room_type` → Podría mantenerse para analizar discrepancias entre la habitación reservada y la asignada, que pueden correlacionar con reviews negativas.

💡 **Conclusión:**

El dataset está bien estructurado y, tras limpieza y transformación, queda listo para análisis exploratorio avanzado.

---
## `reviews_raw`

In [5]:
# Llamamos a la función eda_preliminar guardada en el archivo de sorporte
eda_preliminar(reviews_raw)



👀 Visualización de las primeras y últimas filas



,Hotel_Address,Additional_Number_of_Scoring,Review_Date,Average_Score,Hotel_Name,Reviewer_Nationality,Negative_Review,Review_Total_Negative_Word_Counts,Total_Number_of_Reviews,Positive_Review,Review_Total_Positive_Word_Counts,Total_Number_of_Reviews_Reviewer_Has_Given,Reviewer_Score,Tags,days_since_review,lat,lng
0,s Gravesandestraat 55 Oost 1092 AA Amsterdam ...,194,8/3/2017,7.7,Hotel Arena,Russia,I am so angry that i made this post available...,397,1403,Only the park outside of the hotel was beauti...,11,7,2.9,"[' Leisure trip ', ' Couple ', ' Duplex Double...",0 days,52.360576,4.915968
1,s Gravesandestraat 55 Oost 1092 AA Amsterdam ...,194,8/3/2017,7.7,Hotel Arena,Ireland,No Negative,0,1403,No real complaints the hotel was great great ...,105,7,7.5,"[' Leisure trip ', ' Couple ', ' Duplex Double...",0 days,52.360576,4.915968
2,s Gravesandestraat 55 Oost 1092 AA Amsterdam ...,194,7/31/2017,7.7,Hotel Arena,Australia,Rooms are nice but for elderly a bit difficul...,42,1403,Location was good and staff were ok It is cut...,21,9,7.1,"[' Leisure trip ', ' Family with young childre...",3 days,52.360576,4.915968


,Hotel_Address,Additional_Number_of_Scoring,Review_Date,Average_Score,Hotel_Name,Reviewer_Nationality,Negative_Review,Review_Total_Negative_Word_Counts,Total_Number_of_Reviews,Positive_Review,Review_Total_Positive_Word_Counts,Total_Number_of_Reviews_Reviewer_Has_Given,Reviewer_Score,Tags,days_since_review,lat,lng
515735,Wurzbachgasse 21 15 Rudolfsheim F nfhaus 1150 ...,168,8/19/2015,8.1,Atlantis Hotel Vienna,Egypt,The ac was useless It was a hot week in vienn...,19,2823,No Positive,0,3,2.5,"[' Leisure trip ', ' Family with older childre...",715 day,48.203745,16.335677
515736,Wurzbachgasse 21 15 Rudolfsheim F nfhaus 1150 ...,168,8/17/2015,8.1,Atlantis Hotel Vienna,Mexico,No Negative,0,2823,The rooms are enormous and really comfortable...,25,3,8.8,"[' Leisure trip ', ' Group ', ' Standard Tripl...",717 day,48.203745,16.335677
515737,Wurzbachgasse 21 15 Rudolfsheim F nfhaus 1150 ...,168,8/9/2015,8.1,Atlantis Hotel Vienna,Hungary,I was in 3rd floor It didn t work Free Wife,13,2823,staff was very kind,6,1,8.3,"[' Leisure trip ', ' Family with young childre...",725 day,48.203745,16.335677




------------------------------------------------------------------------


🎲 Visualización de filas aleatorias



,Hotel_Address,Additional_Number_of_Scoring,Review_Date,Average_Score,Hotel_Name,Reviewer_Nationality,Negative_Review,Review_Total_Negative_Word_Counts,Total_Number_of_Reviews,Positive_Review,Review_Total_Positive_Word_Counts,Total_Number_of_Reviews_Reviewer_Has_Given,Reviewer_Score,Tags,days_since_review,lat,lng
240145,7 rue du G n ral de Larminat 15th arr 75015 Pa...,103,6/9/2016,8.8,Hotel Ares Eiffel,Canada,The bed for the 3rd person in the room was a ...,26,667,No Positive,0,5,7.5,"[' Leisure trip ', ' Family with older childre...",420 day,48.850328,2.298202
31826,10b Chestnut Plaza Westfield Stratford City Ol...,248,1/22/2016,9.2,Staybridge Suites London Stratford,United Kingdom,Quite a troublesome location if returning lat...,28,945,great room with excellent facilities lovey br...,11,5,8.8,"[' Leisure trip ', ' Couple ', ' One Bedroom S...",559 day,51.542635,-0.007327
76637,18 Avenue De Suffren Entr e au 22 rue Jean Rey...,428,3/1/2016,8.6,Pullman Paris Tour Eiffel,United Kingdom,No Negative,0,2764,Bed was very comfortable view was perfect fro...,36,1,10.0,"[' Leisure trip ', ' Couple ', ' Deluxe King R...",520 day,48.855566,2.292624




------------------------------------------------------------------------


📏 Dimensión del DataFrame

Número de filas del DataFrame: 515738
Número de columnas del DataFrame: 17


------------------------------------------------------------------------


🧾 Información general

<class 'pandas.DataFrame'>
RangeIndex: 515738 entries, 0 to 515737
Data columns (total 17 columns):
 #   Column                                      Non-Null Count   Dtype  
---  ------                                      --------------   -----  
 0   Hotel_Address                               515738 non-null  str    
 1   Additional_Number_of_Scoring                515738 non-null  int64  
 2   Review_Date                                 515738 non-null  str    
 3   Average_Score                               515738 non-null  float64
 4   Hotel_Name                                  515738 non-null  str    
 5   Reviewer_Nationality                        515738 non-null  str    
 6   Negative_Review         

Index(['Hotel_Address', 'Additional_Number_of_Scoring', 'Review_Date',
       'Average_Score', 'Hotel_Name', 'Reviewer_Nationality',
       'Negative_Review', 'Review_Total_Negative_Word_Counts',
       'Total_Number_of_Reviews', 'Positive_Review',
       'Review_Total_Positive_Word_Counts',
       'Total_Number_of_Reviews_Reviewer_Has_Given', 'Reviewer_Score', 'Tags',
       'days_since_review', 'lat', 'lng'],
      dtype='str')



------------------------------------------------------------------------


🔢 Estadísticas descriptivas de las variables numéricas



,count,mean,std,min,25%,50%,75%,max
Additional_Number_of_Scoring,515738.0,498.082,500.538,1.000,169.000,341.000,660.000,2682.000
Average_Score,515738.0,8.397,0.548,5.200,8.100,8.400,8.800,9.800
Review_Total_Negative_Word_Counts,515738.0,18.539,29.691,0.000,2.000,9.000,23.000,408.000
Total_Number_of_Reviews,515738.0,2743.744,2317.465,43.000,1161.000,2134.000,3613.000,16670.000
Review_Total_Positive_Word_Counts,515738.0,17.776,21.804,0.000,5.000,11.000,22.000,395.000
Total_Number_of_Reviews_Reviewer_Has_Given,515738.0,7.166,11.040,1.000,1.000,3.000,8.000,355.000
Reviewer_Score,515738.0,8.395,1.638,2.500,7.500,8.800,9.600,10.000
lat,512470.0,49.442,3.466,41.328,48.215,51.500,51.516,52.400
lng,512470.0,2.824,4.579,-0.370,-0.143,0.011,4.834,16.429




------------------------------------------------------------------------


🔡 Estadísticas descriptivas de las variables categóricas



,count,unique,top,freq
Hotel_Address,515738,1493,163 Marsh Wall Docklands Tower Hamlets London ...,4789
Review_Date,515738,731,8/2/2017,2585
Hotel_Name,515738,1492,Britannia International Hotel Canary Wharf,4789
Reviewer_Nationality,515738,227,United Kingdom,245246
Negative_Review,515738,330011,No Negative,127890
Positive_Review,515738,412601,No Positive,35946
Tags,515738,55242,"[' Leisure trip ', ' Couple ', ' Double Room '...",5101
days_since_review,515738,731,1 days,2585


### **Informe de EDA preliminar – Dataset `reviews_raw`**

**1. Información general del Dataset**

- El dataset contiene 515738 registros y 17 variables.

- Presenta pocos valores nulos en lat y lng (~0.63%).

- Existen algunos duplicados, pero la estructura general es consistente.

**2. Hallazgos por tipo de variable**

Variables numéricas:

- Reviewer_Score y Average_Score: media ≈ 8.4 → alta satisfacción general.
- Total_Number_of_Reviews: gran dispersión → diferentes niveles de popularidad.

Variables de texto y categóricas:

- Positive_Review y Negative_Review: alta variabilidad en contenido.
- Reviewer_Nationality: predominan usuarios del Reino Unido.

**3. Transformaciones clave**

Estandarización:

- Nombres de las columnas → minúsculas + snake_case.
- reviewer_nationality: “United Kingdom” → “UK”.
- tags: limpiar caracteres innecesarios.

Tipos de datos:

- review_date → datetime.
- days_since_review → eliminar la palabra “day” y convertir a int.

Feature engineering:

- hotel_address → crear city y country.
- Reducir cada review en positive_review y negative_review a una “palabra clave”

Eliminación:

. review_total_negative_word_counts y review_total_positive_word_counts.
- lat y lng: redundancia, ya tenemos la localización.

💡 Conclusión:

El dataset es rico en información y adecuado para análisis de sentimiento. Tras limpieza, estandarización y creación de nuevas variables, queda optimizado para análisis avanzado e integración con bookings.

In [6]:
reviews_raw['country'] = reviews_raw['Hotel_Address'].apply(lambda x: x.split()[-1])
reviews_raw.sample(5)

,Hotel_Address,Additional_Number_of_Scoring,Review_Date,Average_Score,Hotel_Name,Reviewer_Nationality,Negative_Review,Review_Total_Negative_Word_Counts,Total_Number_of_Reviews,Positive_Review,Review_Total_Positive_Word_Counts,Total_Number_of_Reviews_Reviewer_Has_Given,Reviewer_Score,Tags,days_since_review,lat,lng,country
45012,130 135 Minories City of London London EC3N 1N...,465,8/1/2016,8.9,The Chamberlain,Australia,Everything was excellent,5,2543,Everything was top class this hotel really li...,51,21,10.0,"[' Leisure trip ', ' Couple ', ' Luxury Double...",367 day,51.512246,-0.075733,Kingdom
170905,39 Buckingham Palace Road Westminster Borough ...,332,8/30/2016,8.7,Rubens At The Palace,United Kingdom,Room was very small no prices on the restaura...,23,1449,The breakfast was off the chart,8,1,9.2,"[' Business trip ', ' Solo traveler ', ' Class...",338 day,51.498147,-0.143649,Kingdom
32358,11 12 Poplar Mews Shepherd s Bush Hammersmith ...,227,10/3/2015,8.0,Shepherd s Bush Boutique Hotel,United Kingdom,Internet did not revealed true location which...,48,1069,Looked great on internet nothing else,8,4,4.6,"[' Business trip ', ' Couple ', ' King Room ',...",670 day,51.505454,-0.226222,Kingdom
394740,Pallars 128 130 Sant Mart 08018 Barcelona Spain,252,5/29/2017,7.9,Sall s Hotel Pere IV,United Kingdom,The location of the hotel was a little far aw...,25,3130,The staff were extremely nice and helpful and...,34,5,9.2,"[' Leisure trip ', ' Couple ', ' Deluxe Double...",66 days,41.396927,2.192267,Spain
396209,Parkring 12 01 Innere Stadt 1010 Vienna Austria,251,5/6/2017,9.0,Hotel Am Parkring,United Arab Emirates,Furniture slightly old,5,2304,Excellent location Walking distance to old to...,37,6,8.3,"[' Leisure trip ', ' Solo traveler ', ' Double...",89 days,48.205633,16.377862,Austria


In [7]:
reviews_raw['city'] = reviews_raw['Hotel_Address'].apply(lambda x: x.split()[-2])
reviews_raw.sample(5)

,Hotel_Address,Additional_Number_of_Scoring,Review_Date,Average_Score,Hotel_Name,Reviewer_Nationality,Negative_Review,Review_Total_Negative_Word_Counts,Total_Number_of_Reviews,Positive_Review,Review_Total_Positive_Word_Counts,Total_Number_of_Reviews_Reviewer_Has_Given,Reviewer_Score,Tags,days_since_review,lat,lng,country,city
72214,173 Gloucester Place Westminster Borough Londo...,468,7/11/2017,7.7,Park Avenue Baker Street,United Kingdom,Fire alarm testing late afternoon without war...,8,1913,Comfortable modern room,4,14,7.9,"[' Leisure trip ', ' Couple ', ' Deluxe Double...",23 days,51.524145,-0.160991,Kingdom,United
150520,33 37 Hogarth Road Kensington and Chelsea Lond...,989,7/10/2016,8.4,Park Grand London Kensington,United Kingdom,No Negative,0,4660,Convenient for London Underground Clean comfo...,14,8,8.3,"[' Business trip ', ' Group ', ' Superior Doub...",389 day,51.493847,-0.191758,Kingdom,United
93464,2 Wallenberg Place Westminster Borough London ...,687,10/18/2015,8.9,The Montcalm Marble Arch,Italy,Service very slow and inappropriate I ve been...,33,2752,Not noisy,3,4,4.6,"[' Business trip ', ' Solo traveler ', ' Montc...",655 day,51.515052,-0.159239,Kingdom,United
371640,Linzer Stra e 165 14 Penzing 1140 Vienna Austria,185,5/11/2017,8.0,Hotel Kavalier,Czech Republic,We were waiting 30 minutes for order in hotel...,51,3333,We announced to come with small child and chi...,45,5,7.1,"[' Leisure trip ', ' Family with young childre...",84 days,48.194049,16.297183,Austria,Vienna
479004,Via Laietana 30 Ciutat Vella 08003 Barcelona S...,153,10/1/2016,8.9,Grand Hotel Central,United Kingdom,No Negative,0,988,Bed so comphy Pool bar and cocktails were fab...,25,1,10.0,"[' Leisure trip ', ' Couple ', ' Executive Dou...",306 day,41.384969,2.177663,Spain,Barcelona
